tests/installation/leap2sle_zdup.pm

```diff
@@ -0,0 +1,55 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Those are steps for leap to sle zypper dup. Includes: switch to a tty, full patch
+# leap system, install SUSEConnect, Register at SCC to get SLE repo, List and disable all
+# openSUSE repo, add modules need for installation, Migrate installed packages to SLES repo,
+# Remove orphaned packages, reboot the system
+# Maintainer: Yutao Wang <yuwang@suse.com>
+
+use base "installbasetest";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use registration;
+use version_utils 'is_desktop_installed';
+use power_action_utils 'power_action';
+
+sub run {
+    my ($self) = @_;
+
+    $self->wait_boot(textmode => !is_desktop_installed(), bootloader_time => 300, ready_time => 600);
+
+    select_console('root-console');
+    # Create a snapshot with specified description to do snapper rollback
+    if (get_var('HDDVERSION', '') !~ /13.1/) {
+        assert_script_run "snapper create --type pre --cleanup-algorithm=number --print-number --userdata important=yes --description 'b_zdup migration'";
+    }
+
+    systemctl 'set-default --force multi-user.target';
+    # The CD was ejected in the bootloader test
+    type_string("/sbin/reboot\n");
+
+    reset_consoles;
+    select_console('root-console');
+    fully_patch_system();
+    zypper_call 'in SUSEConnect';
+    assert_script_run "SUSEConnect -r " . get_var("SCC_REGCODE") . " -p SLES/15.2/x86_64";
```

Comment:
why calling `registration::register_product` below? but here you're doing a manual call? (I'd suggest to stick to the same way (register_product)

Reply:
If use 'register_product', it only has SUSEConnect -r regcode. It will register opensuse system. I must ass ' -p SLES/15.2/x86_64' to register SLES. But I change this part to use 'add_suseconnect_product' instead of amanual call

Reply:
@foursixnine  Do you have any other suggestion? If no, please approve this PR. Thanks a lot.